examples/with-algolia-react-instantsearch/utils/index.ts

```diff
@@ -0,0 +1,10 @@
+import type { SearchState } from 'react-instantsearch-core'
+import qs from 'qs'
+
+export const createURL = (state: SearchState) => `?${qs.stringify(state)}`
+
+export const pathToSearchState = (path: string) =>
+  path.includes('?') ? qs.parse(path.substring(path.indexOf('?') + 1)) : {}
+
+export const searchStateToURL = (searchState: SearchState) =>
+  searchState ? `${window.location.pathname}?${qs.stringify(searchState)}` : ''
```

Comment:
Could we just use `URL` and `URLSearchParams` instead and get rid of `qs`?

Reply:
@balazsorban44 I looked into it, and unfortunately `URLSearchParams` does not support nesting, which `react-instantsearch` search state does: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams#parameters